core/src/main/java/com/amplifyframework/util/Empty.java

```diff
@@ -49,9 +48,9 @@ public static boolean check(@Nullable Map<?, ?> map) {
      * Returns true if str is null or empty.
      *
      * TextUtils.isEmpty() provides the same behavior, but throws an exception when called from a unit test, since the
-     *  Android library is not available in unit tests.  Empty.check() is preferred to allow for unit testing.
-     *
-     * @see <a href="http://tools.android.com/tech-docs/unit-testing-support#TOC-Method-...-not-mocked.">documentation</a> for more details.
+     *  Android library is not available in unit tests.  Empty.check() is preferred to allow for unit testing.  See
+     *  <a href="http://tools.android.com/tech-docs/unit-testing-support#TOC-Method-...-not-mocked.">Android
+     *  tools documentation</a> for details.
```

Comment:
The "not mocked" error can be fixed by using `@RunWith(RobolectricTestRunner.class)`.

The Robolectric test runner includes a version of the Android APIs that can run on the host JVM -- not the phone's ART/JVM.

The downside is that Robolectric tests are slower.

To make this comment accurate, we should really go through hand update all references to `TextUtils.isEmpty`, and try to strip off the test-runners, to see if the tests will run after the change.

Overall it seems reasonable -- avoiding `TextUtils` basically means faster unit tests (_if_ the source code doesn't use any other Android-y stuff, and so Robolectric annotations could be removed from the class.)

Reply:
Thanks for that explanation!   There were 3 other usages of `TextUtils.isEmpty` in the codebase, and I've updated them to use `Empty.check`.  As far as I can tell, those usages aren't covered by unit tests (unfortunately), so there aren't any related `Robolectric` annotations to be removed.